test/integration/basic_test.go

```diff
@@ -0,0 +1,167 @@
+	"github.com/DATA-DOG/godog/gherkin"
+
+	"github.com/minishift/minishift/test/integration/util"
+)
+
+var lastCommandOutput CommandOutput
+
+type CommandOutput struct {
+	Command  string
+	StdOut   string
+	StdErr   string
+	ExitCode int
+}
+
+type Minishift struct {
+	runner util.MinishiftRunner
+}
+
+func (m *Minishift) shouldHaveState(state string) error {
+	if !m.runner.CheckStatus(state) {
+		return fmt.Errorf("Expected minishift to be: %s", state)
+	}
+
+	return nil
+}
+
+func (m *Minishift) shouldHaveAValidIPAddress() error {
+	ip, _, _ := m.runner.RunCommand("ip")
+	ip = strings.TrimRight(ip, "\n")
+	if net.ParseIP(ip) == nil {
+		return fmt.Errorf("IP command returned an invalid address: %s", ip)
+	}
+
+	return nil
+}
+
+func (m *Minishift) executingCommand(command string) error {
+	// TODO: there must be smarter way to destruct
+	cmdOut, cmdErr, cmdExit := m.runner.RunCommand(command)
+	lastCommandOutput = CommandOutput{
+		command,
+		cmdOut,
+		cmdErr,
+		cmdExit,
+	}
+
+	// Beware: you are responsible to verify the lastCommandOutput!
+	return nil
+}
+
+func compareExpectedWithActualContains(expected string, actual string) error {
+
+	if !strings.Contains(actual, expected) {
+		return fmt.Errorf("Output did not match. Expected: %s, Actual: %s", expected, actual)
+	}
+
+	return nil
+}
+
+func compareExpectedWithActualEquals(expected string, actual string) error {
+
+	if actual != expected {
+		return fmt.Errorf("Output did not match. Expected: %s, Actual: %s", expected, actual)
+	}
+
+	return nil
+}
+
+func selectFieldFromLastOutput(commandField string) string {
+	outputField := ""
+	switch commandField {
+	case "stdout":
+		outputField = lastCommandOutput.StdOut
+	case "stderr":
+		outputField = lastCommandOutput.StdErr
+	case "exitcode":
+		outputField = strconv.Itoa(lastCommandOutput.ExitCode)
+	default:
+		fmt.Errorf("Incorrect field type specified for comparison: %s", commandField)
+	}
+	return outputField
+}
+
+func (m *Minishift) commandReturnShouldContain(commandField string, expected *gherkin.DocString) error {
+	return compareExpectedWithActualContains(expected.Content, selectFieldFromLastOutput(commandField))
+}
+
+func (m *Minishift) commandReturnShouldEqual(commandField string, expected *gherkin.DocString) error {
+	return compareExpectedWithActualEquals(expected.Content, selectFieldFromLastOutput(commandField))
+}
+
+func (m *Minishift) commandReturnShouldBeEmpty(commandField string) error {
+	return compareExpectedWithActualEquals("", selectFieldFromLastOutput(commandField))
+}
+
+func FeatureContext(s *godog.Suite) {
+	var givenArgs = flag.String("minishift-args", "", "Arguments to pass to minishift")
+	var givenPath = flag.String("binary", fmt.Sprintf("../../out/%s-amd64/minishift", runtime.GOOS), "Path to minishift binary")
+
+	runner := util.MinishiftRunner{
+		CommandArgs: *givenArgs,
+		CommandPath: *givenPath}
+
+	m := &Minishift{runner: runner}
+
+	s.Step(`minishift should be "([^"]*)"`, m.shouldHaveState)
```

Comment:
Technically I think it should be "Minishift" (capital 'M'). Same for the IP address case. I think we said we want to use _minishift_ for the cases where we actually refer to the binary itself. Opposed to _Minishift_ as the tool/program/cli.